model_analyzer/config/generate/neighborhood.py

```diff
@@ -228,38 +227,129 @@ def _get_visited_coordinates(self) -> List[Coordinate]:
                 visited_coordinates.append(deepcopy(coordinate))
         return visited_coordinates
 
+    def _get_initialized_coordinates(self) -> List[Coordinate]:
+        initialized_coordinates = []
+        for coordinate in self._neighborhood:
+            if coordinate != self._home_coordinate \
+                    and self._is_coordinate_initialized(coordinate):
+                initialized_coordinates.append(deepcopy(coordinate))
+        return initialized_coordinates
+
     def _get_step_vector(self) -> Coordinate:
         """
         Calculate a vector that indicates a direction to step from the
         home coordinate (current center).
 
+        If the home coordinate is passing the constraints, the method
+        calculates the step vector based on the objectives given.
+        Otherwise, it calculates the step vector based on the constraints
+        so that it steps toward the region that passes the constraints.
+
         Returns
         -------
         step_vector
             a coordinate that tells the direction to move.
         """
-        step_vector = Coordinate([0] * self._config.get_num_dimensions())
-
-        vectors, measurements = self._compile_neighborhood_measurements()
+        vectors, measurements = self._get_constraints_passing_measurements()
```

Comment:
The function name implies that you are getting constraints that pass measurements. Rename to get_measurements_passing_constraints

Reply:
Updated.